csai/src/main/java/com/dice/doris/csai/MainActivity.kt

```diff
@@ -0,0 +1,119 @@
+import com.dice.doris.csai.entity.VideoInfo
+import com.dice.doris.csai.util.SourceUtils
+import com.dice.doris.csai.util.SourceUtils.SourceCallback
+import com.dice.shield.drm.entity.ActionToken
+import com.diceplatform.doris.ExoDoris
+import com.diceplatform.doris.ExoDorisBuilder
+import com.diceplatform.doris.entity.DorisAdEvent
+import com.diceplatform.doris.entity.ImaCsaiProperties
+import com.diceplatform.doris.entity.Source
+import com.diceplatform.doris.entity.SourceBuilder
+import com.diceplatform.doris.entity.TextTrack
+import com.diceplatform.doris.ext.imacsai.ExoDorisImaCsaiBuilder
+import com.diceplatform.doris.ext.imacsailive.ExoDorisImaCsaiLiveBuilder
+import com.diceplatform.doris.ui.ExoDorisPlayerView
+
+class MainActivity : AppCompatActivity(), SourceCallback {
+    private val progressBar: ProgressBar by lazy { findViewById(R.id.progress_bar) }
+    private val playerView: ExoDorisPlayerView by lazy { findViewById(R.id.playerView) }
+    private var player: ExoDoris? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        // Global configuration.
+        Log.setLogLevel(Log.LOG_LEVEL_ALL)
+        playerView.keepScreenOn = true  // Keep device screen always on
+        playerView.setShowPreviousButton(false)
+        playerView.setShowNextButton(false)
+        playerView.setShowSubtitleButton(true)
+        SourceUtils.getSource(this, this)
+    }
+
+    override fun onSourceCallback(videoInfo: VideoInfo, adsConfiguration: AdsConfiguration?) {
+        val imaCsaiProperties = parseCsaiProperties(adsConfiguration)
+
+        val src = SourceBuilder()
+            .setId(CsaiConfig.videoId)
+            .setUrl(videoInfo.url)
+            .setDrmParams(videoInfo.drm?.let { drm -> ActionToken(CsaiConfig.videoId, videoInfo.url, drm.url, drm.jwtToken, "widevine") })
+            .setImaCsaiProperties(imaCsaiProperties)
+            .setTextTracks(parseTextTrack(videoInfo))
+            .build()
+
+        player = createPlayer(Source.getAdType(src))
+
+        player?.addAnalyticsListener(EventLogger())
+
+        player?.addListener(object : Player.Listener {
+            override fun onRenderedFirstFrame() {
+                progressBar.visibility = View.GONE
+            }
+        })
+
+        playerView.player = player?.exoPlayer
+
+        player?.load(src)
+    }
+
+    private fun createPlayer(adType: DorisAdEvent.AdType): ExoDoris {
+        val builder = if (adType === DorisAdEvent.AdType.IMA_CSAI) {
+            ExoDorisImaCsaiBuilder(this@MainActivity).apply {
+                setAdViewProvider(playerView)
+            }
+        } else if (adType === DorisAdEvent.AdType.IMA_CSAI_LIVE) {
+            ExoDorisImaCsaiLiveBuilder(this@MainActivity).apply {
+                setLiveAdSurfaceView(playerView.videoSurfaceView as SurfaceView)
+                setAdViewProvider(playerView)
```

Comment:
For ima csai live player, we need two players, one for content playback, second for ad player.

Reply:
try to fix it with #13